compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1538,7 +1538,8 @@ object Types {
               else recomputeMember(d) // symbol could have been overridden, recompute membership
             else {
               val newd = loadDenot
-              if (newd.exists) newd else d.staleSymbolError
+              if (newd.exists || ctx.mode.is(Mode.Interactive)) newd
```

Comment:
Isn't this going to fail with a NoDenotation error later on? If you really want to return the old denotation this should be `d`, not `newd`